### ISSUE TITLE
Increasing the number of tfrecord data blocks leads to test accuracy loss

### ISSUE THREAD

**Issue by Tron-x** [horovod/horovod#1716]
I am running a distributed training experiment, training Imagenet to use resnet50. When I shard Imagenet into 1024 tfrecord, the experiment is correct, whether I use 1024gpu or 2048gpus. But when I shard Imagenet into 2048 tfreocrd. No matter how many cards I use(8gpus, 1024gpus，2048gpus), the test accuracy will be lost. But I didn't find out what was wrong with reading the data. The training script I use is https://github.com/aws-samples/deep-learning-models/blob/3bcdb4cb0d451c16abbe395e73d713a026d4b458/models/resnet/tensorflow/train_imagenet_resnet_hvd.py#L355
The read data section is
![image](https://user-images.githubusercontent.com/7659413/74088648-8d748500-4ad3-11ea-9526-3f7bdca1d239.png)

**Comment by tgaddair** [comment]
Hey @Tron-x, that's very strange.  I'm not familiar with this implementation of ImageNet training.  Have you tried our [keras_imagenet_resnet50.py](https://github.com/horovod/horovod/blob/master/examples/keras_imagenet_resnet50.py) example?  Does it have similar issues as the tfrecord block size changes?

**Comment by Tron-x** [comment]
Hi, this script mainly implements Lars algorithm and some parameter tuning to ensure the accuracy of large batch training. This problem seems strange. I tried another script, but it didn't appear this problem, but I couldn't find where the script problem occurred。
